src/DispatchListener.php

```diff
@@ -92,12 +93,14 @@ public function onDispatch(MvcEvent $e)
         try {
             $return = $controller->dispatch($request, $response);
         } catch (\Exception $ex) {
-            $e->setError($application::ERROR_EXCEPTION)
-                  ->setController($controllerName)
-                  ->setControllerClass(get_class($controller))
-                  ->setParam('exception', $ex);
-            $results = $events->trigger(MvcEvent::EVENT_DISPATCH_ERROR, $e);
-            $return = $results->last();
+            $e->setName(MvcEvent::EVENT_DISPATCH_ERROR);
+            $e->setError($application::ERROR_EXCEPTION);
+            $e->setController($controllerName);
+            $e->setControllerClass(get_class($controller));
+            $e->setParam('exception', $ex);
+
+            $results = $events->triggerEvent($e);
+            $return  = $results->last();
```

Comment:
Chain which? I purposely didn't chain the setters for the event, as the `EventInterface` no longer defines fluent setters (though `MvcEvent` still does). Or did you mean this?

```php
$return = $events->triggerEvent($e)->last();
```